src/views/catalog-product-details.vue

```diff
@@ -764,7 +765,40 @@ export default defineComponent({
         console.error(error)
       }
     },
-    async updateReserveInvConfig(value: boolean) {
+    async showAlert(header: string, message: string, cancelButtonLabel: string, successButtonLabel: string){
+      const alert = await alertController.create({
+        header: translate(header),
+        message: translate(message),
+        buttons: [{
+          text: translate(cancelButtonLabel),
+          role: 'cancel'
+        },
+        {
+          text: translate(successButtonLabel),
+          role: 'success',
+        }]
+      });
+
+      await alert.present();
+      const { role } = await alert.onDidDismiss();
+      return role === 'success';
+    },
+    async updateReserveInvConfig(event: any) {
+      // For preventing ion-toggle from toggling
+      event.stopImmediatePropagation();
+
+      const isChecked = event.target.checked;
+      const successButtonLabel = isChecked ? 'disable' : 'enable';
+      const alertHeader = isChecked ? 'Disable Reserve Inventory?' : 'Enable Reserve Inventory?';
+      const alertMessage = isChecked ? 'Disabling inventory reservations prevents committed inventory from being reduced until it has been shipped. Orders that are pending allocation or haven’t been shipped will not be reduced from sellable inventory.' : 'Enabling inventory reservations reduces inventory counts for committed inventory before it has been shipped. Committed inventory includes orders waiting to be brokered or waiting to be shipped.';
+
+      if (await this.showAlert(alertHeader, alertMessage, 'Cancel', successButtonLabel)) {
```

Comment:
I think we can skip passing label for cancel button as its always the same. Also instead of calling the variables as `alertHeader` etc, we can call them simply `header` or `message`.